pyanaconda/storage_utils.py

```diff
@@ -415,3 +419,114 @@ def filter_unsupported_disklabel_devices(devices):
+    to force a refresh of the udev cache.
+
+    :param str device_name: name of the device to check
+    :param devicetree: device tree to look up devices in (optional)
+    :type devicetree: :class:`blivet.DeviceTree`
+    :param bool refresh_udev_cache: governs if the udev device cache should be refreshed
+    :returns: True if the device name corresponds to a disk, False if not
+    :rtype: bool
+    """
+    if devicetree is None:
+        global udev_device_dict_cache
+        if device_name:
+            if udev_device_dict_cache is None or refresh_udev_cache:
+                # Lazy load the udev dick that contains the {device_name : udev_device,..,}
+                # mappings. The operation could be quite costly due to udev_settle() calls,
+                # so we cache it in this non-elegant way.
+                # An unfortunate side effect of this is that udev devices that show up after
+                # this function is called for the first time will not be taken into account.
+                udev_device_dict_cache = {udev.device_get_name(d): d for d in udev.get_devices()}
+            udev_device = udev_device_dict_cache.get(device_name)
+            return udev_device and udev.device_is_disk(udev_device)
+        else:
+            return False
+    else:
+        device = devicetree.getDeviceByName(device_name)
+        return device and device.isDisk
+
+def device_matches(spec, devicetree=None, disks_only=False):
+    """Return names of block devices matching the provided specification.
+
+    :param str spec: a device identifier (name, UUID=<uuid>, &c)
+    :keyword devicetree: device tree to look up devices in (optional)
+    :type devicetree: :class:`blivet.DeviceTree`
+    :param bool disks_only: if only disk devices matching the spec should be returned
+    :returns: names of matching devices
+    :rtype: list of str
+
+    The spec can contain multiple "sub specs" delimited by a |, for example:
+
+    "sd*|hd*|vd*"
+
+    In such case we resolve the specs from left to right and return all
+    unique matches, for example:
+
+    ["sda", "sda1", "sda2", "sdb", "sdb1", "vdb"]
+
+    If disks_only is specified we only return
+    disk devices matching the spec. For the example above
+    the output with disks_only=True would be:
+
+    ["sda", "sdb", "vdb"]
+
+    Also note that parse methods will not have access to a devicetree, while execute
+    methods will. The devicetree is superior in that it can resolve md
+    array names and in that it reflects scheduled device removals, but for
+    normal local disks udev.resolve_devspec should suffice.
+    """
+
+    matches = []
+    # the device specifications might contain multiple "sub specs" separated by a |
+    # - the specs are processed from left to right
+    for single_spec in spec.split("|"):
+        full_spec = single_spec
+        if not full_spec.startswith("/dev/"):
+            full_spec = os.path.normpath("/dev/" + full_spec)
```

Comment:
This is, and always has been, a mistake IMO. Users are capable of typing `/dev/` and that saves us from pretending we know whether prepending `/dev/` is a sane thing to do.